examples/guess-the-sketch/src/app.py

```diff
@@ -77,106 +77,135 @@ def agones_health():
-
-        similarity_response_cur = requests.post(f'http://genai-api.genai.svc/genai/text', headers=headers, json=payload_similarity)
-
-        # logger.info(f'Similarity response: {similarity_response_cur.text}')
-        similarity_prompt_cur_without_quota = similarity_response_cur.text.strip('"')
-        logger.info(f'Similarity response without quota: {similarity_prompt_cur_without_quota}')
-
-        parts = similarity_prompt_cur_without_quota.split(",")  # Split the string by commas
-
-        # Extract numbers and convert to float
-        score1 = float(parts[0].split()[-1])  
-        score2 = float(parts[1].split()[-1]) 
-
-        logger.info(f'num1: {score1}')
-        logger.info(f'num2: {score2}')
-
-        if score1 > score2:
-            emit('winner', {'message': 'You won!'}, room=request.sid)
-            emit('winner', {'message': 'You lost!'}, room=partner_id)
-        elif score1 < score2:
-            emit('winner', {'message': 'You lost!'}, room=request.sid)
-            emit('winner', {'message': 'You won!'}, room=partner_id)
-        else:
-            emit('winner', {'message': 'It\'s a tie!'}, room=request.sid)
-            emit('winner', {'message': 'It\'s a tie!'}, room=partner_id)
-
-        del pending_requests[request.sid]
-        del pending_requests[partner_id]
-
-@socketio.on('message')
-def handle_message(message):
-    request_id = request.sid
-    logger.info('Received request %s', request_id)
-    logger.info('socket id %s', request.sid)
-    logger.info('Message: %s', message)
-    logger.info('Pending requests: %s', pending_requests)
-
-    # TODO ... (Timeout logic - Consider a background task or database) ...
-
-    pending_requests[request_id] = {
-        'timestamp': time.time(),
-        'partner': None,
-        'socket_id': request.sid,
-        'message': message,
-        'partner_message': None,
-        'guess': None
+    if player_guess.get(player_id) is None:
+        player_guess[player_id] = {}
+    player_guess[player_id][round] = {
+        'guess': message
+    }
+    
+    # Send next picture
+    if round != game_round:
+        next_round = round + 1
+        while player_prompt[oppontent_id][next_round]['picture'] is None:
+            time.sleep(0.1)
+        emit('llm_response', {'image': player_prompt[oppontent_id][next_round]['picture'], 'opponentId': oppontent_id, 'round': next_round}, room=player_id)
+        logger.info('Sent image %s to %s', str(next_round), player_id)
+
+    # Generate and store the picture for the guess
+    guess_payload = {
+        'prompt': f'''{message}''',
+    }
+    model_response = requests.post(f'http://genai-api.genai.svc/genai/image', headers=headers, json=guess_payload)
+    # stable diffusion
+    #model_response_cur = requests.post(f'http://stable-diffusion-api.genai.svc/generate', headers=headers, json=payload_cur)
+    encoded_image = base64.b64encode(model_response.content).decode('utf-8')
```

Comment:
is .. the web socket not safe for binary data?

Reply:
I was also curious about that when I was looking at websockets. From what I could tell, they can transmit both binary and text data, but I didn't see a good way to mix and match them over the same websocket.